bridges/snowbridge/pallets/ethereum-client/src/lib.rs

```diff
@@ -641,5 +657,29 @@ pub mod pallet {
 			let signing_root = Self::compute_signing_root(header, domain)?;
 			Ok(signing_root)
 		}
+
+		/// Updates are free if the update is successful and the interval between the latest
+		/// finalized header in storage and the newly imported header is large enough. All
+		/// successful sync committee updates are free.
+		pub(super) fn may_refund_call_fee(
+			latest_slot: u64,
+			improved_by_slot: u64,
+			sync_committee_updated: bool,
+		) -> Pays {
+			// If the sync committee was successfully updated, the update may be free.
+			if sync_committee_updated {
+				return Pays::No;
+			}
+
+			// If free headers are allowed and the latest finalized header is larger than the
+			// minimum slot interval, the header import transaction is free.
+			if let Some(free_headers_interval) = T::FreeHeadersInterval::get() {
+				if improved_by_slot >= latest_slot + free_headers_interval as u64 {
```

Comment:
So this is basically saying you get a free update if the update is far enough in the future?

Reply:
Yeah, seems follow the same pattern for the P->K bridge [here](https://github.com/paritytech/polkadot-sdk/blob/dc4047c7d05f593757328394d1accf0eb382d709/bridges/modules/grandpa/src/lib.rs#L131) with free update at a minimal interval.

Reply:
Exactly. It will be set to every 32 slots (so basically all the finalized updates), but if we ever need to make it less frequent we can.